src/VssDatabase.cpp

```diff
@@ -580,6 +588,15 @@ jsoncons::json VssDatabase::getSignal(class WsChannel& channel, const VSSPath& p
         value[spath] = "---";
       }
       valueArray.insert(valueArray.array_range().end(), value);
+      //This will add the "last"  timestamp, changing behavior from previous
+      //"timestamp of the get request" approach 
+      //Both are not very helpful when querying multiple values.
+      //This will be fixed once https://github.com/eclipse/kuksa.val/issues/158
+      //is implemented, as VISS2 will allow attaching individual timestamps to
+      //individual data
+      if (result.contains("timestamp")) {
+        answer["timestamp"] = result["timestamp"].as<string>();
```

Comment:
if we will fix it for #158, why do we need this temp solution? It is totally confused to use the timestamp of the last element!

Reply:
The VISS1 standard demands a timestamp in answers, see the Schema in https://www.w3.org/TR/vehicle-information-service/

Before we had it as "timestamp of get request" (useless but syntactically compliant), now we have he timestamp of "one of" the requested data (also not that useful, but still compliant), Where is is "better", might when querying "ligically connected" data such a GPS position where a whole subtree will be updated at the same time. 
 
Removing timestamp would not be protocol compliant.

Because this problem has been recognized VISS2 change the behaviour, attaching timestamp to data. But that is a larger change, hence a separate ticket

Reply:
pls add openpoints to the issue or create a new issue